### PR TITLE
MM-48609 : channels/DM/GMs  moveTo sub menu options revise

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**
#### Summary
Sub menu items of channel move to of channel menu improved for following cases
- when a sub menu is under channel header dropdown, its shows correct categories i.e For DM/GM category "Channel" is not shown and vice versa
- when multiple same dms/gms are selected, Other categories along with "Direct Messages" is shown and not "Channel" category
- when multiple channels non dm/gm are selected, other categories along with "Channel" is shown and "Direct Messages" is not shown.
- when multiple any types of channels which included mix of DM/GM and other are selected, we show other categories but not show either of "Direct messages" nor "Channels"

Reviewing the PR
- [actual code change commit](https://github.com/mattermost/mattermost-webapp/pull/11694/commits/971dd6ea17f52971f064d4f409b4da1d693c04df)
- Rest is file name change for better knowing what that component is : `category_menu_items` => `channel_move_to_sub_menu`

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-48609

#### Related Pull Requests
n/a

#### Screenshots
tba

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by esethna — APPROVED** [review]
Thanks!

File: components/channel_move_to_sub_menu/index.tsx [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+        const filteredCategories = categories.filter((category) => category.type !== CategoryTypes.CHANNELS && category.type !== CategoryTypes.DIRECT_MESSAGES);
+        return createSubmenuItemsForCategoryArray(filteredCategories);
+    }
+
+    if (!categories) {
+        return null;
+    }
+
+    return (
+        <React.Fragment>
```
Comment (hmhealey) [inline components/channel_move_to_sub_menu/index.tsx:154]: The outer fragment isn't needed by the look of it

**Review by hmhealey — APPROVED** [review]

File: components/channel_move_to_sub_menu/index.tsx (hunk 1 above)
Comment (M-ZubairAhmed) [reply components/channel_move_to_sub_menu/index.tsx]: removed the outer fragment

**Review by m1lt0n — APPROVED** [review]

**Review by jgilliam17 — APPROVED** [review]
Thanks @M-ZubairAhmed 
Tested, looks good to merge.
Verified move to sub menu options when single or multi-selecting sidebar channel items of identical/mixed types.
[E2E report](https://mattermost-cypress-report.s3.amazonaws.com/269457-49d1e5f-onprem-ent-webapp-pr-11694-mm-ee-test:49d1e5f/mochawesome.html) - no PR related failures. I noticed that not all suites ran 1587/1620. If we identify e2e failures after merge I can create follow ups to fix.

**Pull request merged**